packages/@lwc/engine-core/src/testFeatureFlag.ts

```diff
@@ -0,0 +1,19 @@
+import features from '@lwc/features';
+
+if (process.env.NODE_ENV !== 'production') {
+    window.addEventListener('test-dummy-flag', () => {
+        let hasFlag = false;
+        if (features.DUMMY_TEST_FLAG) {
+            hasFlag = true;
+        }
+
+        window.dispatchEvent(
+            new CustomEvent('has-dummy-flag', {
+                detail: {
+                    package: '@lwc/engine-core',
+                    hasFlag,
+                },
+            })
+        );
+    });
+}
```

Comment:
It'd be great if we didn't have to ship this code even in dev mode, but I can't think of a good way to add code that only matters for our Karma tests.

I could also abstract this into a shared utility in `@lwc/shared`, but it seems silly since, again, it's only used in tests.

Reply:
What do you think about adding a compilation-time feature flag that is forever configured to `false`?

For example:
```
// @lwc/engine-dom/src/index.ts
...
import features from '@lwc/features';
if (features.ENABLE_TEST_EXCEPTION) {
  throw new Error('Compile-time processing of feature flags is broken.');
}
...
```
```
// @lwc/features/flags.ts
const features: FeatureFlagMap = {
    ...
    ENABLE_TEST_EXCEPTION: false,
    ...
}
```

Reply:
Yeah I think this is a much simpler solution than mine. Let's do it.

Reply:
Ah wait, except what this doesn't cover is if the features Babel plugin doesn't run at all. In that case, the flag will just silently not work, and no error will be thrown.

Reply:
You know what, maybe that's fine, if a feature flag is not working in `@lwc/engine-dom`, then presumably the test for that flag will fail anyway. Let's go with your solution.

Reply:
Reverted due to some confusion about what the tests were doing.